### PR TITLE
MINOR: supported all_index in search index configuration form

### PR DESCRIPTION
**Pull request by Ashish8689**
Issue #16624

### Describe your changes:

- I worked on supporting the Hierarchy Tree for all the indexes
- Allow `all clear` in the config form
- fix the beta badge color
- fix the checkbox color with the primary color.

Issue : 
image
image

Resolved : 

image
image

https://github.com/open-metadata/OpenMetadata/assets/66266464/c936668c-2c0d-4228-a54b-7c313671cb70

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by mohityadav766 — DISMISSED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/common/Form/JSONSchema/JsonSchemaWidgets/TreeSelectWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  ...rest
+}) => {
+  const treeData = useMemo(
+    () => [
+      {
+        title: 'All',
+        value: 'all',
+        key: 'all',
+        children: rest.options.enumOptions?.map((node) => ({
+          title: node.label,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/common/Form/JSONSchema/JsonSchemaWidgets/TreeSelectWidget.tsx:33]:
```suggestion
          title: startCase(node.label),
```

**Review by chirag-madlani — COMMENTED** [review]
@Ashish8689 I saw bit odd color for re-create index check box can you check once?
It should be primary color

**Comment by Ashish8689** [comment]
@chirag-madlani changes made as per comments!

**Review by chirag-madlani — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/SearchIndexApplication.spec.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+ */
+import test, { expect, Page } from '@playwright/test';
+import { GlobalSettingOptions } from '../../constant/settings';
+import { getApiContext, redirectToHomePage } from '../../utils/common';
+import { settingClick } from '../../utils/sidebar';
+
+// use the admin user to login
+test.use({ storageState: 'playwright/.auth/admin.json' });
+
+const verifyApplicationTriggerToastData = async (page: Page) => {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/SearchIndexApplication.spec.ts:21]: @Ashish8689 , let's move util methods in util file.

File: openmetadata-ui/src/main/resources/ui/src/components/Settings/Services/AddService/Steps/SelectServiceType.tsx [hunk 3]
```diff
@@ -146,11 +144,7 @@ const SelectServiceType = ({
                 {BETA_SERVICES.includes(
                   type as DatabaseServiceType | PipelineServiceType
                 ) ? (
-                  <Badge
-                    className="service-beta-tag"
-                    color={theme.primaryColor}
-                    count={t('label.beta')}
-                  />
+                  <Badge className="service-beta-tag" count={t('label.beta')} />
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Settings/Services/AddService/Steps/SelectServiceType.tsx:147]: @Ashish8689 , what is the reason for not using the theme color?

Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/Settings/Services/AddService/Steps/SelectServiceType.tsx:147]: we already have the styling with the class `service-beta-tag`. so not needed here!

**Review by chirag-madlani — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/SearchIndexApplication.spec.ts (hunk 2 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/SearchIndexApplication.spec.ts:21]: @Sachin-chaurasiya will cover this with new PR, with all the other spec files using the same method!

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**